lib/src/command/remove.dart

```diff
@@ -143,13 +143,13 @@ class RemoveCommand extends PubCommand {
 
         if (dependenciesNode is YamlMap &&
             dependenciesNode.containsKey(package)) {
-          if (dependenciesNode.length == 1) {
+          yamlEditor.remove([dependencyKey, package]);
+          found = true;
+          // Check if the dependencies or dev_dependencies map is now empty
+          // If it is empty, remove the key as well
+          if (dependenciesNode.isEmpty) {
```

Comment:
I don't think the `dependenciesNode` gets mutated by the `yamlEditor.remove([dependencyKey, package]);` above.

You could probably replace this with something like:
```suggestion
          if ((yamlEditor.parseAt([dependencyKey]) as YamlMap).isEmpty) {
```